src/java/org/apache/cassandra/db/commitlog/MemoryMappedSegment.java

```diff
@@ -105,4 +112,27 @@ protected void internalClose()
         FileUtils.clean(buffer);
         super.internalClose();
     }
+
+    protected static class MemoryMappedSegmentBuilder extends CommitLogSegment.Builder<MemoryMappedSegment>
+    {
+        public MemoryMappedSegmentBuilder(AbstractCommitLogSegmentManager segmentManager)
+        {
+            super(segmentManager);
+        }
+
+        @Override
+        public MemoryMappedSegment build()
+        {
+            return new MemoryMappedSegment(segmentManager,
+                                           path ->  FileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE));
+        }
+
+        @Override
+        public SimpleCachedBufferPool createBufferPool()
+        {
+            return new SimpleCachedBufferPool(DatabaseDescriptor.getCommitLogMaxCompressionBuffersInPool(),
+                                              DatabaseDescriptor.getCommitLogSegmentSize(),
+                                              BufferType.ON_HEAP);
```

Comment:
On-heap here is giving a strong "something's wrong" signal. Could we not create a buffer pool i.e. return null in this case?